### DISCUSSION TITLE
Deploy whisper as an AWS Lambda function

### DISCUSSION THREAD

**Discussion by michaelrubinfeldgg** [openai/whisper#1364] [Q&A]
Hi everyone, I'm trying to build a lambda function containing whisper. In my latest attempt I've received the following error message:

No such file or directory: 'ffmpeg'"

Did anyone come across this issue? I have ffmpeg.exe in the project file of the lambda, and when I run the same code outside of a lambda function context (code in .py file and ffmpeg.exe in the same directory) everything works fine.

**Comment by Soykertje** [comment]
If you're using any Debian-based or linux image to run it, you don't need that file, instead install ffmpeg (and all of its dependencies) using the appropriate package manager and it should work.

**Comment by ururk** [comment]
To add to @Soykertje's comment, ffmpeg won't work when run like you have it locally. You need to install a version of ffmpeg that's compatible with AWS Lambda's runtime. Amazon has a guide on how to create a lambda layer:

https://aws.amazon.com/blogs/media/processing-user-generated-content-using-aws-lambda-and-ffmpeg/

Essentially, this bundles/packages ffmpeg such that it is available to your lambda function invocation when added as a layer. The one caveat - ffmpeg won't be in the system path. That guide installs it at this path:

`/opt/bin/ffmpeg`

And so you would need to export `/opt/bin` to the PATH so that `ffmpeg` is discoverable by `ffmpeg-python`

**Comment by loganpowell** [comment]
@michaelrubinfeldgg did you ever get this to work? I've found examples for other models (e.g. [ocr](https://github.com/aws-samples/ml-inference-using-aws-lambda-and-amazon-efs)), which leverage AWS EFS for model caching, but if you have succeeded here, we'd love a repo to reference!

**Comment by ventz** [comment]
Posting this here for anyone interested, Ive had this working for quite some time - here is my repo:

https://github.com/ventz/whisper-openai-container

**Reply by loganpowell** [reply]
Thank you @ventz 🙏

**Reply by loganpowell** [reply]
@ventz can you help me understand what this line does in your `Dockerfile`:

```dockerfile
RUN whisper --model_dir /usr/local --model medium audio >> /dev/null 2>&1; exit 0
```

specifically, what `audio  >> /dev/null 2>&1; exit 0` does...

**Reply by ventz** [reply]
@loganpowell It downloads the model during build time so that Lambdas run quicker (at the expense of storage size)

The section you are asking about redirects stderr to stdout and sends that to dev null. The exit 0 provides a clean exit code. 

This is needed as otherwise the exit code generated from the model download will break the Docker build.

**Reply by loganpowell** [reply]
Perfect, I just encountered that error code :D thank you sir
